tools/c7n_azure/c7n_azure/policy.py

```diff
@@ -104,19 +113,22 @@ def _get_parameters(self, template_util):
         data = self.policy.data
 
         updated_parameters = {
-            'name': (data['mode']['provision-options']['servicePlanName'] +
-                     '-' +
-                     data['name']).replace(' ', '-').lower(),
-
-            'storageName': (data['mode']['provision-options']['servicePlanName']
-                            ).replace('-', '').lower(),
             'dockerVersion': CONST_DOCKER_VERSION,
             'functionsExtVersion': CONST_FUNCTIONS_EXT_VERSION
         }
 
         if 'mode' in data:
             if 'provision-options' in data['mode']:
                 updated_parameters.update(data['mode']['provision-options'])
+                if 'servicePlanName' in data['mode']['provision-options']:
+                    updated_parameters['name'] = (
+                        data['mode']['provision-options']['servicePlanName'] +
+                        '-' + data['name']
+                    ).replace(' ', '-').lower()
```

Comment:
Can we add the StringUtils class a normalize function that does `.replace(' ', '-').lower()`

Reply:
I think that's a great refactor for later! I think it's out of the scope of this PR though.